client/allocrunner/interfaces/task_lifecycle.go

```diff
@@ -43,8 +43,9 @@ type TaskHook interface {
 }
 
 type TaskPrestartRequest struct {
-	// HookData is previously set data by the hook
-	HookData map[string]string
+	// PreviousState is previously set data by the hook. It must be copied
+	// to State below to be maintained across restarts.
+	PreviousState map[string]string
```

Comment:
I renamed the `HookData` field in Request and Response because there was no way you could use them without accidentally using the wrong one at least once (eg setting state on Request instead of Response).

Hopefully the names and comments make the use and behavior more obvious.

Reply:
👍 - this is a nice change for readability